packages/table-core/src/utils/getGroupedRowModel.ts

```diff
@@ -118,6 +118,10 @@ export function getGroupedRowModel<TData extends RowData>(): (
               })
 
               subRows.forEach(subRow => {
+                if (!subRow.parentId) {
+                  Object.assign(subRow, { parentId: id });
```

Comment:
Unfortunately, this also introduces a bug where the table will crash after ungrouping by a column. Needs to be looked into more, but this will be part of the solution.

Reply:
Good catch Kevin. I was so focused on the solution for my current use-case that I forgot about ungrouping 😢

Reply:
Looking into it, we have to find a way to remove the parentId as we ungroup.

Reply:
I'm trying to navigate this code, trying to follow the same approach for how the `.depth` value is assigned/unassigned.. But I see that property being modified in here:

https://github.com/TanStack/table/blob/46b160dfd37917e6cc1f8e53c091345574a0fae4/packages/table-core/src/utils/getGroupedRowModel.ts#L38

So, I have couple questions:

1. What is the difference between https://github.com/TanStack/table/blob/46b160dfd37917e6cc1f8e53c091345574a0fae4/packages/table-core/src/utils/getGroupedRowModel.ts#L38-L41 and the section I touched on this PR? mostly, they seem the same, but not sure why they happen in two diff places
2. Would that the better place to assign/unassign the `parentId` 🤔?

Reply:
😞 well, `depth` is also wrong

When we ungroup the column, they all stay `depth: 1`;

<img width="363" alt="image" src="https://github.com/TanStack/table/assets/9648559/66654650-d014-45e3-9f91-a61d4c2c09a2">

 but they should go back to `depth: 0` which is the default value when ungrouped 🤕

<img width="265" alt="image" src="https://github.com/TanStack/table/assets/9648559/c83b3267-7a93-40c2-b38c-dc97cbec34b2">

Reply:
For what is worth:

assigning/un-assigning `depth` in v7 works just fine, here an example: https://codesandbox.io/s/amazing-water-pcepcv?file=/src/App.js